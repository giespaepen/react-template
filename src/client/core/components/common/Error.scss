@import "../../../assets/style/AppProperties.scss";
@import "../../../assets/style/AppMixins.scss";

.error {
    h2 {
        background-color:lighten($color-error,20%);
        margin-bottom: $bar-height;
        padding:$bar-height/3 $bar-height/4;
    }
    .error-stack{
        font-family: monospace;
        margin-top:$bar-height;
    }
}