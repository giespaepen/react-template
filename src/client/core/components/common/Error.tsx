import "./Error.scss";
import * as React from "react";
import State from "../../state/BaseState";
import { Action } from "redux";
import { Dispatch } from "redux";
import { IErrorProperties } from "./Properties";
import { connect } from "react-redux";
import { createErrorOccurredAction } from "../../actions/ActionCreators";

/**
 * Stateless Error component
 */
export default class Error extends React.Component<IErrorProperties, State> {

    public render(): any {
        return <div className="error">
            <h2>{this.props.code} - Error :(</h2>
            <h3>{this.props.error.message}</h3>
            <p className="error-content">
                Something went wrong while requesting the page.
            </p>
            <p className="error-stack">
                {this.props.error.stack}
            </p>
        </div>;
    }
}
