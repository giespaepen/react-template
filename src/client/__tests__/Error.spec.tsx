import * as React from "react";
import * as renderer from "react-test-renderer";
import Error from "../core/components/common/Error";

test("Error snapshot test", () => {
    const error = { name: "error", message: "the error" };
    const component = renderer.create(<Error error={error} code={400} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
