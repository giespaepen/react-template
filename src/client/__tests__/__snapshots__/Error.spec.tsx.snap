// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error snapshot test 1`] = `
<div
  className="error"
>
  <h2>
    400
     - Error :(
  </h2>
  <h3>
    the error
  </h3>
  <p
    className="error-content"
  >
    Something went wrong while requesting the page.
  </p>
  <p
    className="error-stack"
  />
</div>
`;
